web/src/reducers/activities.js

```diff
@@ -0,0 +1,49 @@
+import { ADD_ACTIVITY, UPDATE_ACTIVITY } from '../actions/activities';
+
+const newActivity = id => ({
+  id,
+  name: '',
+  types: ['HIT'],
+  descShort: '',
+  descLong: '',
+  altApproach: ''
+});
+
+const updateEntry = (state, action) => {
```

Comment:
Do you think using [updeep](https://www.npmjs.com/package/updeep) here might be good?  The code would look something like...

```
return updeep({ byId: { [id]: { [name]: value }}}, state);
```

Reply:
whoa - never heard of it, but i'm intrigued!!

Reply:
updeep is awesome, gonna switch to this going forward -- great tip!